### PR TITLE
NTP-921: Changes for VAT

### PR DESCRIPTION
**Pull request by kieranohanlon**
## Context

Ensure that NTP shows VAT prices inc and exc VAT, especially when comparing prices

## Changes proposed in this pull request

Added in if VAT is included, VAT is excluded or not applicable against the prices.

Add in a new drop down option to toggle between the VAT inclusive
image

And VAT exclusive prices
image

Default the drop down selection to inclusive of VAT

TP details page to have a label stating if VAT does not apply if this is the case, to match the table
image

Currently for VAT exempt companies: 

Currently for VAT inclusive companies:

image

As part of this ticket we will change the way that we store the prices in the NTP database, so that they are exclusive of VAT.

When showing VAT inclusive prices we will add 20% and round to the nearest penny as per:

https://www.gov.uk/hmrc-internal-manuals/vat-trader-records/vatrec12030#:~:text=If%20the%20VAT%20on%20any,it%20should%20be%20rounded%20up .

## Guidance to review

AC1: The VAT information is now displayed against the pricing data

GIVEN a search has been made, the search subjects filters have been removed and some TPs have been added to the shortlist

WHEN the shortlist is displayed

THEN the prices now include if the VAT has been included, state if they exclude VAT (if VAT drop down is used) or if it does not apply (to meet exact wording supplied by content team)

AC2: The Prices are shown inclusive of VAT if this drop down option is selected

GIVEN a search has been made, search subjects have been filtered upon and some TPs have been added to the shortlist

WHEN the shortlist is displayed and the VAT drop down is set to show inclusive figures

THEN the TPs prices are shown inclusive of VAT where applicable.  The companies that do not charge VAT will be showing the exclusive figure.

AC3: The Prices are shown exclusive of VAT if this drop down option is selected

GIVEN a search has been made, search subjects have been filtered upon and some TPs have been added to the shortlist

WHEN the shortlist is displayed and the VAT drop down is set to show exclusive figures

THEN the TPs prices are shown exclusive of VAT.

AC4: The TP details page shows a label if the TP does not charge VAT

GIVEN via the search results or all TPs page

WHEN selecting a TP that does not charge VAT

THEN the pricing section will explicitly state that VAT does not apply (exact wording to be supplied by content)

## Link to Jira ticket

https://dfedigital.atlassian.net/browse/NTP-921

## Things to check

- [x] Title is in the format "NTP-XXX: Short description" where NTP-XXX is the Jira ticket reference
- [x] Code and tests follow the [coding standards](/docs/coding-standards.md)
- [x] `dotnet format` has been run in the repository root
- [x] Test coverage of new code is at least 80%
- [x] All UI related acceptance criteria specified in the ticket have been added to the relevant [cypress test feature file(s)](/UI/cypress/e2e/)
- [x] Unless explicitly mentioned in the ticket, all UI work should have been tested as working on desktop and mobile resolutions with JavaScript on and off
- [ ] Database migrations can be applied to the current codebase in main without causing exceptions
- [ ] Debug logging has been added after all logic decision points
- [x] All [automated testing](/README.md#testing) including accessibility and security has been run against your code
- [x] **PR deployment has been signed off by wider team**

### CODE REVIEW

File: Application/Extensions/DecimalExtensions.cs [hunk 1]
```diff
@@ -2,9 +2,22 @@
...
+    //Current VAT is 20%
+    private const decimal VATRate = 20m;
+
     public static string FormatPrice(this decimal price)
     {
         var str = $"{price:C}";
         return str.EndsWith(".00") ? str[0..^3] : str;
     }
+
+    public static decimal AddVAT(this decimal price)
```
[the remark this answers is not in the document]
Comment (kieranohanlon) [reply Application/Extensions/DecimalExtensions.cs:14]: Agreed - Added Unit Test

File: Application/Extensions/DecimalExtensions.cs [hunk 2]
```diff
@@ -2,9 +2,22 @@
...
+    {
+        return Math.Round(price * ((100 + VATRate) / 100), 2);
+    }
+
+    public static decimal RemoveVAT(this decimal price)
```
[the remark this answers is not in the document]
Comment (kieranohanlon) [reply Application/Extensions/DecimalExtensions.cs:19]: Agreed - Added Unit Test

File: Application.Tests/Extensions/DecimalExtensionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    [InlineData(12.58, 10.48)]
+    [InlineData(12.59, 10.49)]
+    [InlineData(12.60, 10.50)]
+    [InlineData(12.61, 10.51)]
+    [InlineData(12.62, 10.52)]
+    [InlineData(12.63, 10.53)]
+    [InlineData(12.64, 10.53)]
+    public void RemoveVAT_RemovesCorrectPercentage(decimal vatExclusive, decimal vatInclusive)
+    {
+        vatExclusive.RemoveVAT().Should().Be(vatInclusive);
```
Comment (hasan3d) [inline Application.Tests/Extensions/DecimalExtensionsTests.cs:38]: The wording doesn't look right to me. The following makes sense to me ` vatInclusive.RemoveVAT().Should().Be(vatExclusive)`

Comment (kieranohanlon) [reply Application.Tests/Extensions/DecimalExtensionsTests.cs]: Done

**Review by hasan3d — APPROVED** [review]
lgtm

**Pull request merged**